openapi/generator.go

```diff
@@ -1161,3 +1177,21 @@ func fieldNameFromTag(sf reflect.StructField, tagName string) string {
 	}
 	return name
 }
+
+/// parseExampleValue is used to transform the string representation of the example value to the correct type.
+func parseExampleValue(rType reflect.Type, value string) (interface{}, error) {
+	switch rType.Kind() {
+	case reflect.Bool:
+		return strconv.ParseBool(value)
+	case reflect.String:
+		return value, nil
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return strconv.ParseInt(value, 10, rType.Bits())
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
```

Comment:
~~Should also handle `reflect.Uintptr`.~~
Edit: we don't handle it already in  the library, so no point I guess.